content/articles/implement-a-collapsible-view-using-a-cardview/index.md

```diff
@@ -0,0 +1,768 @@
+       android:orientation="vertical"
+       app:layout_constraintGuide_begin="16dp" />
+
+   <androidx.constraintlayout.widget.Guideline
+       android:id="@+id/guideline2"
+       android:layout_width="wrap_content"
+       android:layout_height="wrap_content"
+       android:orientation="vertical"
+       app:layout_constraintGuide_end="16dp" />
+
+   <androidx.constraintlayout.widget.Guideline
+       android:id="@+id/guideline3"
+       android:layout_width="wrap_content"
+       android:layout_height="wrap_content"
+       android:orientation="horizontal"
+       app:layout_constraintGuide_begin="16dp" />
+
+   <androidx.constraintlayout.widget.Guideline
+       android:id="@+id/guideline4"
+       android:layout_width="wrap_content"
+       android:layout_height="wrap_content"
+       android:orientation="horizontal"
+       app:layout_constraintGuide_end="16dp" />
+  
+
+</androidx.constraintlayout.widget.ConstraintLayout>
+```
+#### Step Six: Designing the UI of the collapsed card
+Build the UI of the collapsed card. You are going to build the layout of the collapsed card.
+
+```xml
+<ImageView
+   android:id="@+id/imageView2"
+   android:layout_width="wrap_content"
+   android:layout_height="wrap_content"
+   app:layout_constraintBottom_toTopOf="@+id/guideline4"
+   app:layout_constraintStart_toStartOf="@+id/guideline1"
+   app:layout_constraintTop_toTopOf="@+id/guideline3"
+   app:layout_constraintVertical_bias="0.0"
+   app:srcCompat="@android:drawable/ic_menu_info_details"
+   app:tint="@color/teal_200" />
+
+<TextView
+   android:id="@+id/textView14"
+   android:layout_width="wrap_content"
+   android:layout_height="wrap_content"
+   android:layout_marginStart="8dp"
+   android:text="Why should you drink water?"
+   android:textColor="#00255E"
+   android:textSize="14sp"
+   app:layout_constraintBottom_toBottomOf="@+id/imageView2"
+   app:layout_constraintStart_toEndOf="@+id/imageView2"
+   app:layout_constraintTop_toTopOf="@+id/imageView2"
+   app:layout_constraintVertical_bias="0.5" />
+
+<ImageView
+   android:id="@+id/show"
+   android:layout_width="wrap_content"
+   android:layout_height="wrap_content"
+   android:layout_marginEnd="4dp"
+   android:padding="4dp"
+   app:layout_constraintBottom_toBottomOf="@+id/textView14"
+   app:layout_constraintEnd_toStartOf="@+id/guideline2"
+   app:layout_constraintHorizontal_bias="1.0"
+   app:layout_constraintStart_toEndOf="@+id/textView14"
+   app:layout_constraintTop_toTopOf="@+id/textView14"
+   app:srcCompat="@android:drawable/arrow_down_float"
+   app:tint="@color/teal_200" />
+   ```
+
+ Notice how I constrain the views to the guidelines.
+ 
+When you click the imageView with id `show`, the card view will expand, revealing more details.
+ 
+This is what the layout would look like:
+
+![Collapsed card](/engineering-education/implement-a-collapsible-view-using-a-cardview/collapsed_card.png)
+
+#### Step Seven: Design the UI for the expanded card view
+ 
```

Comment:
Explain this step well and explain which view you are using before displaying the code snippets.